qcodes/dataset/data_set.py

```diff
@@ -1693,6 +1699,92 @@ def _flush_data_to_database(self, block: bool = False) -> None:
             log.debug(f"Waiting for write queue to empty.")
             writer_status.data_write_queue.join()
 
+    def _export_file_name(self, prefix: str, export_type: DataExportType):
+        extension = export_type.value
+        return f"{prefix}{self.run_id}.{extension}"
+
+    def _export_as_netcdf(self, path: str, prefix: str) -> str:
+        """Export data as netcdf to a given path with file prefix"""
+        file_name = self._export_file_name(
+            prefix=prefix, export_type=DataExportType.NETCDF)
+        file_path = os.path.join(path, file_name)
+        xarr_dataset = self.to_xarray_dataset()
+        xarr_dataset.to_netcdf(path=file_path)
+        return path
+
+    def _export_as_csv(self, path: str, prefix: str) -> str:
+        """Export data as csv to a given path with file prefix"""
+        file_name = self._export_file_name(
+            prefix=prefix, export_type=DataExportType.CSV)
+        self.write_data_to_text_file(path=path, single_file=True, single_file_name=file_name)
+        return os.path.join(path, file_name)
+
+    def _export_data(
+        self,
+        export_type: DataExportType,
+        path: str = None,
```

Comment:
We are using mypys no_implicit_optional config so all the parameters that have been set as a default value of None should be types as `Optional[str]` etc